### PR TITLE
HPCC-13917 Make sure that query is only fetched on demand

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman rchapman@hpccsystems.com

### CODE REVIEW

File: common/workunit/workunit.cpp [hunk 1]
```diff
@@ -6847,25 +6848,39 @@ IStringVal& CLocalWUQuery::getQueryText(IStringVal &str) const
 
 IStringVal& CLocalWUQuery::getQueryShortText(IStringVal &str) const
 {
-    const char * text = p->queryProp("Text");
```
Comment (richardkchapman) [inline common/workunit/workunit.cpp]: These changes are trying to make the Dali version more efficient - it won't actually make a lot of difference in Cassandra as yet (unless I make the query text even more lazy loaded - which I may)

**Comment by richardkchapman** [comment]
@ghalliday Please review

**Comment by ghalliday** [comment]
@richardkchapman shouldn't this have code to save/restore @ isArchive in the cassandra mappings?

**Comment by richardkchapman** [comment]
@ghalliday It will be saved (along with all the other attributes) by:

    {"attributes", "map<text, text>", "", attributeMapColumnMapper},

**Pull request merged**